services/scan/src/precinct_scanner_state_machine.ts

```diff
@@ -0,0 +1,854 @@
+      type: 'INTERPRETATION_INVALID',
+      interpretation,
+      reason: 'invalid_test_mode',
+    };
+  }
+
+  if (
+    frontType === 'InvalidPrecinctPage' ||
+    backType === 'InvalidPrecinctPage'
+  ) {
+    return {
+      type: 'INTERPRETATION_INVALID',
+      interpretation,
+      reason: 'invalid_precinct',
+    };
+  }
+
+  if (frontType === 'UnreadablePage' || backType === 'UnreadablePage') {
+    return {
+      type: 'INTERPRETATION_INVALID',
+      interpretation,
+      reason: 'unreadable',
+    };
+  }
+
+  // TODO what does this case actually mean?
+  if (
+    frontType === 'UninterpretedHmpbPage' ||
+    backType === 'UninterpretedHmpbPage'
+  ) {
+    return {
+      type: 'INTERPRETATION_INVALID',
+      interpretation,
+      reason: 'unknown',
+    };
+  }
+
+  if (frontType === 'InterpretedBmdPage' || backType === 'InterpretedBmdPage') {
+    return { type: 'INTERPRETATION_VALID', interpretation };
+  }
+
+  assert(
+    frontType === 'InterpretedHmpbPage' && backType === 'InterpretedHmpbPage'
+  );
+  const frontAdjudication = front.interpretation.adjudicationInfo;
+  const backAdjudication = back.interpretation.adjudicationInfo;
+
+  if (
+    frontAdjudication.requiresAdjudication ||
+    backAdjudication.requiresAdjudication
+  ) {
+    const frontReasons = frontAdjudication.enabledReasonInfos;
+    const backReasons = backAdjudication.enabledReasonInfos;
+
+    let reasons: AdjudicationReasonInfo[];
+    // If both sides are blank, the ballot is blank
+    if (
+      frontReasons.length === 1 &&
+      frontReasons[0].type === AdjudicationReason.BlankBallot &&
+      backReasons.length === 1 &&
+      backReasons[0].type === AdjudicationReason.BlankBallot
+    ) {
+      reasons = [{ type: AdjudicationReason.BlankBallot }];
+    }
+    // Otherwise, we can ignore blank sides
+    else {
+      reasons = [...frontReasons, ...backReasons].filter(
+        (reason) => reason.type !== AdjudicationReason.BlankBallot
+      );
+    }
+
+    // If there are any non-blank reasons, they should be reviewed
+    if (reasons.length > 0) {
+      return {
+        type: 'INTERPRETATION_NEEDS_REVIEW',
+        interpretation,
+        reasons,
+      };
+    }
+  }
+
+  return {
+    type: 'INTERPRETATION_VALID',
+    interpretation,
+  };
+}
+
+async function accept({ client, store, interpretation }: Context) {
+  assert(store);
+  assert(client);
+  assert(interpretation);
+  // Optimistically record the interpretation in the store. If accept fails
+  // (e.g. due to paper jam), we delete the interpretation from the store. If
+  // the Plustek crashes during accept, it will automatically accept on reboot.
+  const { sheetId, sheet } = interpretation.interpretation;
+  storeInterpretedSheet(store, sheetId, sheet);
+  debug('Stored sheet before accepting: %s', sheetId);
+  debug('Accepting');
+  const acceptResult = await client.accept();
+  debug('Accept result: %o', acceptResult);
+  acceptResult.unsafeUnwrap();
```

Comment:
is there a missing `return` here?

Reply:
Technically no because we don't actually do anything with the results. `unsafeUnwrap` just makes sure we throw an error if there is one. But I will add the `return` for consistency with the others.

(Makes me realize that one of the downsides of having `Result` type is that whenever you interface with a library that expects thrown errors you have to unwrap)